userspace/libsinsp/container_engine/docker/podman.cpp

```diff
@@ -153,8 +135,33 @@ int detect_podman(const sinsp_threadinfo *tinfo, std::string &container_id)
 		return 0; // root
 	}
 
-	std::string systemd_cgroup = get_systemd_cgroup(tinfo);
-	return get_podman_cgroup_uid(systemd_cgroup, container_id, tinfo->m_tid);
+	// the kernel driver does not return cgroups without subsystems (e.g. name=systemd)
+	// in the cgroups field, so we have to do a check here, and load /proc/pid/cgroups
+	// ourselves if needed
+	if(tinfo->get_cgroup("name=systemd", cgroup))
+	{
+		return get_podman_cgroup_uid(cgroup, container_id, tinfo->m_tid);
+	}
+
+	sinsp_threadinfo proc_cgroups_tinfo;
+	proc_cgroups_tinfo.m_tid = tinfo->m_tid;
+	sinsp_cgroup::instance().lookup_cgroups(proc_cgroups_tinfo);
+	const auto& cgroups = proc_cgroups_tinfo.cgroups();
+	if(cgroups.empty())
+	{
+		return libsinsp::procfs_utils::NO_MATCH;
+	}
```

Comment:
This is actually unneeded, isn't it? When empty the for loop will exit immediately and return NO_MATCH